rust/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs

```diff
@@ -98,6 +119,10 @@ impl From<FallbackError> for ProviderError {
     }
 }
 
+fn warn_span(idx: usize, provider: &PrometheusJsonRpcClient<Http>) {
+    let _span = warn_span!("request_with_fallback", provider_index=%idx, ?provider).entered();
```

Comment:
I don't think this does anything since the span is not returned. Lmk if I am wrong.

Also because of how the macros work you should always invoke them where they are being made since they include line numbers. If you do keep a wrapper fn-like use a macro.